casper/src/main/scala/coop/rchain/casper/api/GraphGenerator.scala

```diff
@@ -23,90 +24,87 @@ object GraphzGenerator {
 
   type ValidatorsBlocks = Map[Long, List[ValidatorBlock]]
 
-  final case class DagInfo[G[_]](
+  final case class DagInfo(
       validators: Map[String, ValidatorsBlocks] = Map.empty,
       timeseries: List[Long] = List.empty
   )
 
   object DagInfo {
-    def empty[G[_]]: DagInfo[G] = DagInfo[G]()
+    def empty: DagInfo = DagInfo()
```

Comment:
Defining default values directly on a class makes it harder to use and read when instance is created. So we should avoid doing that and instead create separate functions like _empty_ here which will create instance with defaults or define their own default parameters.

```scala
def empty: DagInfo = DagInfo(validators = Map.empty, timeseries = List.empty)
```

Reply:
Resolved in [commit](https://github.com/rchain/rchain/pull/3574/commits/280fdf4f0422d1a3c443b9eb27b675842fc6a472). @tgrospic, can you explain please why it is harder to use and read when the defaults are provided directly in class definition?